scipy/stats/tests/test_stats.py

```diff
@@ -5875,94 +5944,126 @@ def test_ttest_ind_nan_2nd_arg():
+@array_api_compatible
+@pytest.mark.skip_xp_backends(np_only=True,
+                              reasons=["Other backends don't like integers"])
+@pytest.mark.usefixtures("skip_xp_backends")
+def test_gh5686(xp):
+    mean1, mean2 = xp.asarray([1, 2]), xp.asarray([3, 4])
+    std1, std2 = xp.asarray([5, 3]), xp.asarray([4, 5])
+    nobs1, nobs2 = xp.asarray([130, 140]), xp.asarray([100, 150])
     # This will raise a TypeError unless gh-5686 is fixed.
     stats.ttest_ind_from_stats(mean1, std1, nobs1, mean2, std2, nobs2)
 
 
-def test_ttest_ind_from_stats_inputs_zero():
+@array_api_compatible
+@pytest.mark.skip_xp_backends(cpu_only=True,
+                              reasons=['Uses NumPy for pvalue, CI'])
+@pytest.mark.usefixtures("skip_xp_backends")
+def test_ttest_ind_from_stats_inputs_zero(xp):
     # Regression test for gh-6409.
-    result = stats.ttest_ind_from_stats(0, 0, 6, 0, 0, 6, equal_var=False)
-    assert_equal(result, [np.nan, np.nan])
+    zero = xp.asarray(0.)
+    six = xp.asarray(6.)
+    NaN = xp.asarray(xp.nan)
+    res = stats.ttest_ind_from_stats(zero, zero, six, zero, zero, six, equal_var=False)
+    xp_assert_equal(res.statistic, NaN)
+    xp_assert_equal(res.pvalue, NaN)
 
 
-def test_ttest_single_observation():
+@array_api_compatible
+@pytest.mark.skip_xp_backends(cpu_only=True,
+                              reasons=['Uses NumPy for pvalue, CI'])
+@pytest.mark.usefixtures("skip_xp_backends")
+def test_ttest_uniform_pvalues(xp):
     # test that p-values are uniformly distributed under the null hypothesis
     rng = np.random.default_rng(246834602926842)
-    x = rng.normal(size=(10000, 2))
-    y = rng.normal(size=(10000, 1))
+    x = xp.asarray(rng.normal(size=(10000, 2)))
+    y = xp.asarray(rng.normal(size=(10000, 1)))
     q = rng.uniform(size=100)
 
     res = stats.ttest_ind(x, y, equal_var=True, axis=-1)
-    assert stats.ks_1samp(res.pvalue, stats.uniform().cdf).pvalue > 0.1
-    assert_allclose(np.percentile(res.pvalue, q*100), q, atol=1e-2)
+    pvalue = np.asarray(res.pvalue)
+    assert stats.ks_1samp(pvalue, stats.uniform().cdf).pvalue > 0.1
+    assert_allclose(np.quantile(pvalue, q), q, atol=1e-2)
 
     res = stats.ttest_ind(y, x, equal_var=True, axis=-1)
-    assert stats.ks_1samp(res.pvalue, stats.uniform().cdf).pvalue > 0.1
-    assert_allclose(np.percentile(res.pvalue, q*100), q, atol=1e-2)
+    pvalue = np.asarray(res.pvalue)
+    assert stats.ks_1samp(pvalue, stats.uniform().cdf).pvalue > 0.1
+    assert_allclose(np.quantile(pvalue, q), q, atol=1e-2)
 
     # reference values from R:
     # options(digits=16)
     # t.test(c(2, 3, 5), c(1.5), var.equal=TRUE)
-    res = stats.ttest_ind([2, 3, 5], [1.5], equal_var=True)
-    assert_allclose(res, (1.0394023007754, 0.407779907736), rtol=1e-10)
+    x, y = xp.asarray([2, 3, 5]), xp.asarray([1.5])
+
+    res = stats.ttest_ind(x, y, equal_var=True)
+    rtol = 1e-6 if is_torch(xp) else 1e-10
```

Comment:
Can you add a comment on why pytorch requires a lower tolerance? I am guessing because `x` is an integral type so in pytorch is cast to float32?

Reply:
`1e-10` was used as the tolerance in the original test, so I wanted to preserve, but `torch` is tested with its default dtype of `float32`, so it can't possibly meet that tolerance.